In [6]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
from sklearn import linear_model, datasets

In [7]:
df_foursquare = pd.read_csv('foursquare_data.csv')
df_yelp = pd.read_csv('yeld_data.csv')
df_merged = pd.read_csv('merged_df.csv')

Build a regression model.

In [8]:
X = df_merged[['Number of Parks found (FourSquare)', 'Distance to Closest Park (FourSquare)', 'Number of Parks found (Yelp)', 'Distance to Closest Park (Yelp)']]
y = df_merged['latitude']

In [9]:
X = sm.add_constant(X) # adding a constant
lin_reg = sm.OLS(y,X)

Provide model output and an interpretation of the results. 

In [10]:
model = lin_reg.fit()
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:               latitude   R-squared:                       0.692
Model:                            OLS   Adj. R-squared:                  0.384
Method:                 Least Squares   F-statistic:                     2.246
Date:                Wed, 22 Nov 2023   Prob (F-statistic):              0.226
Time:                        17:43:03   Log-Likelihood:                 37.610
No. Observations:                   9   AIC:                            -65.22
Df Residuals:                       4   BIC:                            -64.23
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                                            coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------

/Users/yousufellithy/anaconda3/lib/python3.11/site-packages/scipy/stats/_stats_py.py:1806: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=9
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


Is there a correleation between the distnace and number of parks to planes of latitude (of bikes)? The only correlation found was to the Distance to Closest Park from (FourSquare), which had a pvalue of 0.053 (rounded down would be 0.05) and the R-squared: 0.692 which is strong enough. The rest seem to indicate that they have no effect on the planes of latitude of bike stations. 

# Stretch

How can you turn the regression model into a classification model?